components/services/bat_ledger/bat_ledger_impl.cc

```diff
@@ -476,4 +476,67 @@ void BatLedgerImpl::GetOneTimeTips(GetOneTimeTipsCallback callback) {
       BatLedgerImpl::OnGetOneTimeTips, holder, _1, _2));
 }
 
+// static
+void BatLedgerImpl::OnGetActivityInfoList(
+    CallbackHolder<GetActivityInfoListCallback>* holder,
+    const ledger::PublisherInfoList& list,
+    uint32_t num) {
+  std::vector<std::string> json_list;
+  for (auto const& item : list) {
+    json_list.push_back(item.ToJson());
+  }
+
+  if (holder->is_valid()) {
+    std::move(holder->get()).Run(json_list, num);
+  }
+
+  delete holder;
+}
+
+void BatLedgerImpl::GetActivityInfoList(
+    uint32_t start,
+    uint32_t limit,
+    const std::string& json_filter,
+    GetActivityInfoListCallback callback) {
+  auto* holder = new CallbackHolder<GetActivityInfoListCallback>(
+      AsWeakPtr(), std::move(callback));
+
+  ledger::ActivityInfoFilter filter;
+  if (filter.loadFromJson(json_filter)) {
```

Comment:
Why are we converting an ActivityInfoFilter to json in rewards_service only to convert it back to an ActivityInfoFilter here?

Reply:
mojo can only do basic types

Reply:
See above. If we need a different filter, we should probably be calling a different method which is what we would be doing in RewardsService above otherwise.

Reply:
Resolved in comment above